lesson_7/edmond_sabou/lesson7_1.py

```diff
@@ -0,0 +1,23 @@
+import random
+
+
+def uppsercase(func):
+
+    annotations = ["-", "*", "+", "-", "="]
+    annotate = random.choice(annotations)
+
+    def formatting(*args):
+        print(annotate * 20)
+        print(func(*args).upper())
+        print(annotate * 20)
+
+    return formatting
+
+
+@uppsercase
+def print_message(name):
+    return "  Greetings {}!".format(name)
+
+
+print_message("World")
```

Comment:
If you try the following:
```python
msg = print_message('test')
print(msg)
```
Would this still work as intended?

Reply:
It would still work due to the print function included in the decorator, but the `print(msg)` would return a `None` value because I have not included a `return` statement for the `func(*args).upper()` from the decorator.
To enhance this I would replace the `print(func(*args).upper())` with  `return func(*args).upper()` which would eliminate the possibility of ever returning a `None` value.
Ooh and I'll also lose one of the annotations because of it being placed after the `return` statement. You win some lose some.

Reply:
Well, you could test these assumptions by removing the prints from the inner function of the decorator